extension/spring-cloud-stream/src/main/kotlin/io/holunda/camunda/bpm/correlate/ingress/cloudstream/SpringCloudStreamChannelProxyFactory.kt

```diff
@@ -0,0 +1,52 @@
+package io.holunda.camunda.bpm.correlate.ingress.cloudstream
+
+import io.holunda.camunda.bpm.correlate.ingress.ChannelConfigurationProperties
+import io.holunda.camunda.bpm.correlate.ingress.ChannelMessageAcceptor
+import io.holunda.camunda.bpm.correlate.ingress.IngressMetrics
+import io.holunda.camunda.bpm.correlate.ingress.cloudstream.SpringCloudStreamChannelConfiguration.Companion.CHANNEL_TYPE
+import io.holunda.camunda.bpm.correlate.util.getQualifiedBeanWithFallback
+import mu.KLogging
+import org.springframework.beans.factory.InitializingBean
+import org.springframework.context.ApplicationContext
+import org.springframework.context.ApplicationContextAware
+import org.springframework.context.support.GenericApplicationContext
+import java.util.function.Supplier
+
+class SpringCloudStreamChannelProxyFactory(
+  private val channelMessageAcceptor: ChannelMessageAcceptor,
+  private val metrics: IngressMetrics,
+  channelConfigurations: Map<String, ChannelConfigurationProperties>,
+) : ApplicationContextAware, InitializingBean {
+
+  companion object : KLogging()
+
+  private lateinit var applicationContext: GenericApplicationContext
+  private val springCloudConfigurations: Map<String, ChannelConfigurationProperties> by lazy {
+    channelConfigurations.filter { it.value.type == CHANNEL_TYPE && it.value.enabled }
+  }
+
+  override fun setApplicationContext(applicationContext: ApplicationContext) {
+    this.applicationContext = applicationContext as GenericApplicationContext
+  }
+
+  override fun afterPropertiesSet() {
+    if (this::applicationContext.isInitialized) {
```

Comment:
Interesting that this works. As far as I understand, it is very dependent on bean initialization order. Spring cannot know that you are going to register `StreamByteMessageConsumer` beans here. So it cannot automatically make sure this bean is initialized before any `StreamByteMessageConsumer`s are injected anywhere. I wonder how stable this is...

Reply:
are initializing beans not processed before all the others?